x-pack/plugins/actions/server/actions_config.ts

```diff
@@ -93,10 +93,20 @@ function getProxySettingsFromConfig(config: ActionsConfig): undefined | ProxySet
     proxyBypassHosts: arrayAsSet(config.proxyBypassHosts),
     proxyOnlyHosts: arrayAsSet(config.proxyOnlyHosts),
     proxyHeaders: config.proxyHeaders,
-    proxyRejectUnauthorizedCertificates: config.proxyRejectUnauthorizedCertificates,
+    proxyTLSSettings: getTLSSettingsFromConfig(
+      config.tls?.proxyVerificationMode,
+      config.proxyRejectUnauthorizedCertificates
+    ),
   };
 }
 
+function getTLSSettingsFromConfig(
+  verificationMode?: 'none' | 'certificate' | 'full',
+  rejectUnauthorized?: boolean
+): TLSSettings {
+  return { legacyRejectUnauthorized: rejectUnauthorized, verificationMode };
+}
```

Comment:
So with the old config options:
* setting `rejectUnauthorized: true` is equivalent to `verificationMode: 'full'`
* setting `rejectUnauthorized: false` is equivalent to `verificationMode: 'none'`

If you change the `TLSSettings` interface to remove the `rejectUnauthorized` field:

```diff
diff --git a/x-pack/plugins/actions/server/types.ts b/x-pack/plugins/actions/server/types.ts
index 0ac13a986dc..c8c9967afca 100644
--- a/x-pack/plugins/actions/server/types.ts
+++ b/x-pack/plugins/actions/server/types.ts
@@ -152,8 +152,4 @@ export interface ResponseSettings {
 
 export interface TLSSettings {
   verificationMode?: 'none' | 'certificate' | 'full';
-  /**
-   * @deprecated
-   */
-  legacyRejectUnauthorized?: boolean;
 }
```

Then you can change this function to just return `verificationMode` if the `rejectUnauthorized` field is defined:

```suggestion
function getTLSSettingsFromConfig(
  verificationMode?: 'none' | 'certificate' | 'full',
  rejectUnauthorized?: boolean
): TLSSettings {
  if (verificationMode) {
    return { verificationMode };
  } else if (rejectUnauthorized !== undefined) {
    return { verificationMode: rejectUnauthorized ? 'full' : 'none' };
  }
  return {};
}
```

That has some other knock-on effects, for `get_node_tls_options.ts` for example. WDYT?

Reply:
Sounds great! I will do the proper changes